[C,test/client.js,1083]                 server.connection();
[C,test/client.js,1084]                 server.register({ register: Nes, options: { auth: false } }, (err) => {
[C,test/client.js,1085] 
[C,test/client.js,1086]                     expect(err).to.not.exist();
[C,test/client.js,1087] 
[C,test/client.js,1088]                     server.start((err) => {
[C,test/client.js,1089] 
[A,test/client.js,1090]                         expect(err).to.not.exist();
[C,test/client.js,1091]                         const client = new Nes.Client('http://localhost:' + server.info.port);
[C,test/client.js,1092]                         client.connect(() => {
[C,test/client.js,1093] 
[C,test/client.js,1094]                             const handler = function (update) {
[C,test/client.js,1095] 
[C,test/client.js,1096]                                 throw new Error('Must not be called');
[C,test/client.js,1097]                             };
[C,test/client.js,1098] 
[C,test/client.js,1099]                             client.subscribe('/', handler, (err) => {
[C,test/client.js,1100] 
[A,test/client.js,1101]                                 expect(err).to.exist();
[C,test/client.js,1102]                                 client.unsubscribe('/', handler);
[C,test/client.js,1103] 
[C,test/client.js,1104]                                 setTimeout(() => {
[C,test/client.js,1105] 
[C,test/client.js,1106]                                     client.disconnect();
[C,test/client.js,1107]                                     server.stop(done);
[C,test/client.js,1108]                                 }, 20);
[C,test/client.js,1109]                             });
[C,test/client.js,1110]                         });
[C,test/client.js,1111]                     });
[C,test/client.js,1112]                 });
[C,test/client.js,1113]             });
[C,test/client.js,1114] 
[C,test/client.js,1115]             it('subscribes and unsubscribes to a path before connecting', (done) => {
[C,test/client.js,1116] 
@@ -1113,30 +1151,31 @@ describe('Browser', () => {
[C,test/client.js,1151]                 done();
[C,test/client.js,1152]             });
[C,test/client.js,1153] 
[C,test/client.js,1154]             it('errors on subscribe fail', (done) => {
[C,test/client.js,1155] 
[C,test/client.js,1156]                 const server = new Hapi.Server();
[C,test/client.js,1157]                 server.connection();
[C,test/client.js,1158]                 server.register({ register: Nes, options: { auth: false } }, (err) => {
[C,test/client.js,1159] 
[C,test/client.js,1160]                     expect(err).to.not.exist();
[C,test/client.js,1161] 
[C,test/client.js,1162]                     server.subscription('/');
[C,test/client.js,1163] 
[C,test/client.js,1164]                     server.start((err) => {
[C,test/client.js,1165] 
[A,test/client.js,1166]                         expect(err).to.not.exist();
[C,test/client.js,1167]                         const client = new Nes.Client('http://localhost:' + server.info.port);
[C,test/client.js,1168]                         client.connect(() => {
[C,test/client.js,1169] 
[C,test/client.js,1170]                             client._ws.send = function () {
[C,test/client.js,1171] 
[C,test/client.js,1172]                                 throw new Error('failed');
[C,test/client.js,1173]                             };
[C,test/client.js,1174] 
[C,test/client.js,1175]                             client.subscribe('/', Hoek.ignore, (err) => {
[C,test/client.js,1176] 
[C,test/client.js,1177]                                 expect(err).to.exist();
[C,test/client.js,1178]                                 expect(err.message).to.equal('failed');
[C,test/client.js,1179] 
[C,test/client.js,1180]                                 client.disconnect();
[C,test/client.js,1181]                                 server.stop(done);
@@ -1170,30 +1209,31 @@ describe('Browser', () => {
[C,test/client.js,1209]                 });
[C,test/client.js,1210]             });
[C,test/client.js,1211] 
[C,test/client.js,1212]             it('subscribes, unsubscribes, then subscribes again to a path', (done) => {
[C,test/client.js,1213] 
[C,test/client.js,1214]                 const server = new Hapi.Server();
[C,test/client.js,1215]                 server.connection();
[C,test/client.js,1216]                 server.register({ register: Nes, options: { auth: false } }, (err) => {
[C,test/client.js,1217] 
[C,test/client.js,1218]                     expect(err).to.not.exist();
[C,test/client.js,1219] 
[C,test/client.js,1220]                     server.subscription('/', {});
[C,test/client.js,1221] 
[C,test/client.js,1222]                     server.start((err) => {
[C,test/client.js,1223] 
[A,test/client.js,1224]                         expect(err).to.not.exist();
[C,test/client.js,1225]                         const client = new Nes.Client('http://localhost:' + server.info.port);
[C,test/client.js,1226]                         client.connect(() => {
[C,test/client.js,1227] 
[C,test/client.js,1228]                             const handler1 = (update1) => {
[C,test/client.js,1229] 
[C,test/client.js,1230]                                 expect(client.subscriptions()).to.deep.equal(['/']);
[C,test/client.js,1231]                                 expect(update1).to.equal('abc');
[C,test/client.js,1232] 
[C,test/client.js,1233]                                 client.unsubscribe('/');
[C,test/client.js,1234] 
[C,test/client.js,1235]                                 const handler2 = (update2) => {
[C,test/client.js,1236] 
[C,test/client.js,1237]                                     expect(client.subscriptions()).to.deep.equal(['/']);
[C,test/client.js,1238]                                     expect(update2).to.equal('def');
[C,test/client.js,1239]                                     client.disconnect();
@@ -1275,56 +1315,58 @@ describe('Browser', () => {
[C,test/client.js,1315]             });
[C,test/client.js,1316]         });
[C,test/client.js,1317] 
[C,test/client.js,1318]         describe('_beat()', () => {
[C,test/client.js,1319] 
[C,test/client.js,1320]             it('disconnects when server fails to ping', (done) => {
[C,test/client.js,1321] 
[C,test/client.js,1322]                 const server = new Hapi.Server();
[C,test/client.js,1323]                 server.connection();
[C,test/client.js,1324]                 server.register({ register: Nes, options: { auth: false, heartbeat: { interval: 20, timeout: 10 } } }, (err) => {
[C,test/client.js,1325] 
[C,test/client.js,1326]                     expect(err).to.not.exist();
[C,test/client.js,1327] 
[C,test/client.js,1328]                     server.start((err) => {
[C,test/client.js,1329] 
[A,test/client.js,1330]                         expect(err).to.not.exist();
[C,test/client.js,1331]                         const client = new Nes.Client('http://localhost:' + server.info.port);
[C,test/client.js,1332]                         client.onDisconnect = function () {
[C,test/client.js,1333] 
[C,test/client.js,1334]                             server.stop(done);
[C,test/client.js,1335]                         };
[C,test/client.js,1336] 
[C,test/client.js,1337]                         client.connect((err) => {
[C,test/client.js,1338] 
[C,test/client.js,1339]                             expect(err).to.not.exist();
[C,test/client.js,1340] 
[C,test/client.js,1341]                             clearTimeout(server.connections[0].plugins.nes._listener._heartbeat);
[C,test/client.js,1342]                         });
[C,test/client.js,1343]                     });
[C,test/client.js,1344]                 });
[C,test/client.js,1345]             });
[C,test/client.js,1346] 
[C,test/client.js,1347]             it('disconnects when server fails to ping (after a few pings)', (done) => {
[C,test/client.js,1348] 
[C,test/client.js,1349]                 const server = new Hapi.Server();
[C,test/client.js,1350]                 server.connection();
[C,test/client.js,1351]                 server.register({ register: Nes, options: { auth: false, heartbeat: { interval: 20, timeout: 10 } } }, (err) => {
[C,test/client.js,1352] 
[C,test/client.js,1353]                     expect(err).to.not.exist();
[C,test/client.js,1354] 
[C,test/client.js,1355]                     server.start((err) => {
[C,test/client.js,1356] 
[A,test/client.js,1357]                         expect(err).to.not.exist();
[C,test/client.js,1358]                         const client = new Nes.Client('http://localhost:' + server.info.port);
[C,test/client.js,1359]                         client.onDisconnect = function () {
[C,test/client.js,1360] 
[C,test/client.js,1361]                             server.stop(done);
[C,test/client.js,1362]                         };
[C,test/client.js,1363] 
[C,test/client.js,1364]                         client.connect((err) => {
[C,test/client.js,1365] 
[C,test/client.js,1366]                             expect(err).to.not.exist();
[C,test/client.js,1367] 
[C,test/client.js,1368]                             setTimeout(() => {
[C,test/client.js,1369] 
[C,test/client.js,1370]                                 clearTimeout(server.connections[0].plugins.nes._listener._heartbeat);
[C,test/client.js,1371]                             }, 50);
[C,test/client.js,1372]                         });
@@ -30,30 +30,31 @@ describe('register()', () => {
[C,test/index.js,30]         server.register({ register: Nes, options: { auth: false, headers: ['Content-Type'] } }, (err) => {
[C,test/index.js,31] 
[C,test/index.js,32]             expect(err).to.not.exist();
[C,test/index.js,33] 
[C,test/index.js,34]             server.route({
[C,test/index.js,35]                 method: 'GET',
[C,test/index.js,36]                 path: '/',
[C,test/index.js,37]                 handler: function (request, reply) {
[C,test/index.js,38] 
[C,test/index.js,39]                     return reply('hello');
[C,test/index.js,40]                 }
[C,test/index.js,41]             });
[C,test/index.js,42] 
[C,test/index.js,43]             server.start((err) => {
[C,test/index.js,44] 
[A,test/index.js,45]                 expect(err).to.not.exist();
[C,test/index.js,46]                 const client = new Nes.Client('http://localhost:' + server.info.port);
[C,test/index.js,47]                 client.connect(() => {
[C,test/index.js,48] 
[C,test/index.js,49]                     client.request('/', (err, payload, statusCode, headers) => {
[C,test/index.js,50] 
[C,test/index.js,51]                         expect(err).to.not.exist();
[C,test/index.js,52]                         expect(payload).to.equal('hello');
[C,test/index.js,53]                         expect(statusCode).to.equal(200);
[C,test/index.js,54]                         expect(headers).to.deep.equal({ 'content-type': 'text/html; charset=utf-8' });
[C,test/index.js,55] 
[C,test/index.js,56]                         client.disconnect();
[C,test/index.js,57]                         server.stop(done);
[C,test/index.js,58]                     });
[C,test/index.js,59]                 });
[C,test/index.js,60]             });
@@ -76,30 +77,31 @@ describe('register()', () => {
[C,test/index.js,77]         server.register({ register: Nes, options: { onConnection: onConnection, auth: false } }, (err) => {
[C,test/index.js,78] 
[C,test/index.js,79]             expect(err).to.not.exist();
[C,test/index.js,80] 
[C,test/index.js,81]             server.route({
[C,test/index.js,82]                 method: 'GET',
[C,test/index.js,83]                 path: '/',
[C,test/index.js,84]                 handler: function (request, reply) {
[C,test/index.js,85] 
[C,test/index.js,86]                     return reply('hello');
[C,test/index.js,87]                 }
[C,test/index.js,88]             });
[C,test/index.js,89] 
[C,test/index.js,90]             server.start((err) => {
[C,test/index.js,91] 
[A,test/index.js,92]                 expect(err).to.not.exist();
[C,test/index.js,93]                 client = new Nes.Client('http://localhost:' + server.info.port);
[C,test/index.js,94]                 client.connect(() => { });
[C,test/index.js,95]             });
[C,test/index.js,96]         });
[C,test/index.js,97]     });
[C,test/index.js,98] 
[C,test/index.js,99]     it('calls onDisconnection callback', (done) => {
[C,test/index.js,100] 
[C,test/index.js,101]         const server = new Hapi.Server();
[C,test/index.js,102]         server.connection();
[C,test/index.js,103] 
[C,test/index.js,104]         let client;
[C,test/index.js,105]         const onDisconnection = function (ws) {
[C,test/index.js,106] 
[C,test/index.js,107]             expect(ws).to.exist();
@@ -110,24 +112,25 @@ describe('register()', () => {
[C,test/index.js,112]         server.register({ register: Nes, options: { onDisconnection: onDisconnection, auth: false } }, (err) => {
[C,test/index.js,113] 
[C,test/index.js,114]             expect(err).to.not.exist();
[C,test/index.js,115] 
[C,test/index.js,116]             server.route({
[C,test/index.js,117]                 method: 'GET',
[C,test/index.js,118]                 path: '/',
[C,test/index.js,119]                 handler: function (request, reply) {
[C,test/index.js,120] 
[C,test/index.js,121]                     return reply('hello');
[C,test/index.js,122]                 }
[C,test/index.js,123]             });
[C,test/index.js,124] 
[C,test/index.js,125]             server.start((err) => {
[C,test/index.js,126] 
[A,test/index.js,127]                 expect(err).to.not.exist();
[C,test/index.js,128]                 client = new Nes.Client('http://localhost:' + server.info.port);
[C,test/index.js,129]                 client.connect(() => {
[C,test/index.js,130] 
[C,test/index.js,131]                     client.disconnect();
[C,test/index.js,132]                 });
[C,test/index.js,133]             });
[C,test/index.js,134]         });
[C,test/index.js,135]     });
[C,test/index.js,136] });
@@ -24,110 +24,114 @@ const expect = Code.expect;
[C,test/listener.js,24] 
[C,test/listener.js,25] describe('Listener', () => {
[C,test/listener.js,26] 
[C,test/listener.js,27]     describe('_beat()', () => {
[C,test/listener.js,28] 
[C,test/listener.js,29]         it('disconnects client after timeout', (done) => {
[C,test/listener.js,30] 
[C,test/listener.js,31]             const server = new Hapi.Server();
[C,test/listener.js,32]             server.connection();
[C,test/listener.js,33]             server.register({ register: Nes, options: { auth: false, heartbeat: { interval: 20, timeout: 10 } } }, (err) => {
[C,test/listener.js,34] 
[C,test/listener.js,35]                 expect(err).to.not.exist();
[C,test/listener.js,36] 
[C,test/listener.js,37]                 server.start((err) => {
[C,test/listener.js,38] 
[A,test/listener.js,39]                     expect(err).to.not.exist();
[C,test/listener.js,40]                     const client = new Nes.Client('http://localhost:' + server.info.port);
[C,test/listener.js,41]                     client.onDisconnect = function () {
[C,test/listener.js,42] 
[C,test/listener.js,43]                         server.stop(done);
[C,test/listener.js,44]                     };
[C,test/listener.js,45] 
[C,test/listener.js,46]                     client.connect((err) => {
[C,test/listener.js,47] 
[C,test/listener.js,48]                         expect(err).to.not.exist();
[C,test/listener.js,49]                         expect(client._heartbeatTimeout).to.equal(30);
[C,test/listener.js,50] 
[C,test/listener.js,51]                         client._onMessage = function () { };
[C,test/listener.js,52]                     });
[C,test/listener.js,53]                 });
[C,test/listener.js,54]             });
[C,test/listener.js,55]         });
[C,test/listener.js,56] 
[C,test/listener.js,57]         it('disables heartbeat', (done) => {
[C,test/listener.js,58] 
[C,test/listener.js,59]             const server = new Hapi.Server();
[C,test/listener.js,60]             server.connection();
[C,test/listener.js,61]             server.register({ register: Nes, options: { auth: false, heartbeat: false } }, (err) => {
[C,test/listener.js,62] 
[C,test/listener.js,63]                 expect(err).to.not.exist();
[C,test/listener.js,64] 
[C,test/listener.js,65]                 server.start((err) => {
[C,test/listener.js,66] 
[A,test/listener.js,67]                     expect(err).to.not.exist();
[C,test/listener.js,68]                     const client = new Nes.Client('http://localhost:' + server.info.port);
[C,test/listener.js,69]                     client.connect((err) => {
[C,test/listener.js,70] 
[C,test/listener.js,71]                         expect(err).to.not.exist();
[C,test/listener.js,72]                         expect(client._heartbeatTimeout).to.be.false();
[C,test/listener.js,73] 
[C,test/listener.js,74]                         client.disconnect();
[C,test/listener.js,75]                         server.stop(done);
[C,test/listener.js,76]                     });
[C,test/listener.js,77]                 });
[C,test/listener.js,78]             });
[C,test/listener.js,79]         });
[C,test/listener.js,80]     });
[C,test/listener.js,81] 
[C,test/listener.js,82]     describe('broadcast()', () => {
[C,test/listener.js,83] 
[C,test/listener.js,84]         it('sends message to all clients', (done) => {
[C,test/listener.js,85] 
[C,test/listener.js,86]             const server = new Hapi.Server();
[C,test/listener.js,87]             server.connection();
[C,test/listener.js,88]             server.register({ register: Nes, options: { auth: false } }, (err) => {
[C,test/listener.js,89] 
[C,test/listener.js,90]                 expect(err).to.not.exist();
[C,test/listener.js,91] 
[C,test/listener.js,92]                 server.start((err) => {
[C,test/listener.js,93] 
[A,test/listener.js,94]                     expect(err).to.not.exist();
[C,test/listener.js,95]                     const client = new Nes.Client('http://localhost:' + server.info.port);
[C,test/listener.js,96]                     client.onUpdate = function (message) {
[C,test/listener.js,97] 
[C,test/listener.js,98]                         expect(message).to.equal('hello');
[C,test/listener.js,99]                         client.disconnect();
[C,test/listener.js,100]                         server.stop(done);
[C,test/listener.js,101]                     };
[C,test/listener.js,102] 
[C,test/listener.js,103]                     client.connect((err) => {
[C,test/listener.js,104] 
[C,test/listener.js,105]                         expect(err).to.not.exist();
[C,test/listener.js,106]                         server.broadcast('hello');
[C,test/listener.js,107]                     });
[C,test/listener.js,108]                 });
[C,test/listener.js,109]             });
[C,test/listener.js,110]         });
[C,test/listener.js,111] 
[C,test/listener.js,112]         it('sends message to all clients (non participating connections)', (done) => {
[C,test/listener.js,113] 
[C,test/listener.js,114]             const server = new Hapi.Server();
[C,test/listener.js,115]             server.connection();
[C,test/listener.js,116]             server.register({ register: Nes, options: { auth: false } }, (err) => {
[C,test/listener.js,117] 
[C,test/listener.js,118]                 expect(err).to.not.exist();
[C,test/listener.js,119] 
[C,test/listener.js,120]                 server.start((err) => {
[C,test/listener.js,121] 
[A,test/listener.js,122]                     expect(err).to.not.exist();
[C,test/listener.js,123]                     const client = new Nes.Client('http://localhost:' + server.info.port);
[C,test/listener.js,124]                     client.onUpdate = function (message) {
[C,test/listener.js,125] 
[C,test/listener.js,126]                         expect(message).to.equal('hello');
[C,test/listener.js,127]                         client.disconnect();
[C,test/listener.js,128]                         server.stop(done);
[C,test/listener.js,129]                     };
[C,test/listener.js,130] 
[C,test/listener.js,131]                     client.connect((err) => {
[C,test/listener.js,132] 
[C,test/listener.js,133]                         expect(err).to.not.exist();
[C,test/listener.js,134]                         server.connection();
[C,test/listener.js,135]                         server.broadcast('hello');
[C,test/listener.js,136]                     });
[C,test/listener.js,137]                 });
@@ -140,59 +144,61 @@ describe('Listener', () => {
[C,test/listener.js,144]             let client;
[C,test/listener.js,145]             server.connection();
[C,test/listener.js,146]             server.register({ register: Nes, options: { auth: false } }, (err) => {
[C,test/listener.js,147] 
[C,test/listener.js,148]                 expect(err).to.not.exist();
[C,test/listener.js,149] 
[C,test/listener.js,150]                 server.on('log', (event, tags) => {
[C,test/listener.js,151] 
[C,test/listener.js,152]                     expect(event.data).to.equal('update');
[C,test/listener.js,153]                     client.disconnect();
[C,test/listener.js,154]                     server.stop(done);
[C,test/listener.js,155]                 });
[C,test/listener.js,156] 
[C,test/listener.js,157]                 server.start((err) => {
[C,test/listener.js,158] 
[A,test/listener.js,159]                     expect(err).to.not.exist();
[C,test/listener.js,160]                     client = new Nes.Client('http://localhost:' + server.info.port);
[C,test/listener.js,161]                     client.connect((err) => {
[C,test/listener.js,162] 
[C,test/listener.js,163]                         expect(err).to.not.exist();
[C,test/listener.js,164]                         const a = { b: 1 };
[C,test/listener.js,165]                         a.c = a;                    // Circular reference
[C,test/listener.js,166] 
[C,test/listener.js,167]                         server.broadcast(a);
[C,test/listener.js,168]                     });
[C,test/listener.js,169]                 });
[C,test/listener.js,170]             });
[C,test/listener.js,171]         });
[C,test/listener.js,172]     });
[C,test/listener.js,173] 
[C,test/listener.js,174]     describe('subscription()', () => {
[C,test/listener.js,175] 
[C,test/listener.js,176]         it('ignores non participating connections', (done) => {
[C,test/listener.js,177] 
[C,test/listener.js,178]             const server = new Hapi.Server();
[C,test/listener.js,179]             server.connection();
[C,test/listener.js,180]             server.register({ register: Nes, options: { auth: false } }, (err) => {
[C,test/listener.js,181] 
[C,test/listener.js,182]                 expect(err).to.not.exist();
[C,test/listener.js,183]                 server.connection();
[C,test/listener.js,184] 
[C,test/listener.js,185]                 server.subscription('/');
[C,test/listener.js,186] 
[C,test/listener.js,187]                 server.start((err) => {
[C,test/listener.js,188] 
[A,test/listener.js,189]                     expect(err).to.not.exist();
[C,test/listener.js,190]                     const client = new Nes.Client('http://localhost:' + server.connections[0].info.port);
[C,test/listener.js,191]                     client.connect(() => {
[C,test/listener.js,192] 
[C,test/listener.js,193]                         const handler = (update) => {
[C,test/listener.js,194] 
[C,test/listener.js,195]                             expect(update).to.equal('heya');
[C,test/listener.js,196]                             client.disconnect();
[C,test/listener.js,197]                             server.stop(done);
[C,test/listener.js,198]                         };
[C,test/listener.js,199] 
[C,test/listener.js,200]                         client.subscribe('/', handler, (err) => {
[C,test/listener.js,201] 
[C,test/listener.js,202]                             expect(err).to.not.exist();
[C,test/listener.js,203]                             server.publish('/', 'heya');
[C,test/listener.js,204]                         });
@@ -218,30 +224,31 @@ describe('Listener', () => {
[C,test/listener.js,224]                 expect(socket).to.exist();
[C,test/listener.js,225]                 expect(path).to.equal('/');
[C,test/listener.js,226]                 server.stop(done);
[C,test/listener.js,227]             };
[C,test/listener.js,228] 
[C,test/listener.js,229]             server.connection();
[C,test/listener.js,230]             server.register({ register: Nes, options: { auth: false } }, (err) => {
[C,test/listener.js,231] 
[C,test/listener.js,232]                 expect(err).to.not.exist();
[C,test/listener.js,233]                 server.connection();
[C,test/listener.js,234] 
[C,test/listener.js,235]                 server.subscription('/', { onSubscribe: onSubscribe, onUnsubscribe: onUnsubscribe });
[C,test/listener.js,236] 
[C,test/listener.js,237]                 server.start((err) => {
[C,test/listener.js,238] 
[A,test/listener.js,239]                     expect(err).to.not.exist();
[C,test/listener.js,240]                     client = new Nes.Client('http://localhost:' + server.connections[0].info.port);
[C,test/listener.js,241]                     client.connect(() => {
[C,test/listener.js,242] 
[C,test/listener.js,243]                         client.subscribe('/', Hoek.ignore, Hoek.ignore);
[C,test/listener.js,244]                     });
[C,test/listener.js,245]                 });
[C,test/listener.js,246]             });
[C,test/listener.js,247]         });
[C,test/listener.js,248] 
[C,test/listener.js,249]         it('removes subscription notification by path', (done) => {
[C,test/listener.js,250] 
[C,test/listener.js,251]             const server = new Hapi.Server();
[C,test/listener.js,252]             let client;
[C,test/listener.js,253] 
[C,test/listener.js,254]             const onSubscribe = function (socket, path) {
@@ -256,55 +263,57 @@ describe('Listener', () => {
[C,test/listener.js,263]                 expect(socket).to.exist();
[C,test/listener.js,264]                 expect(path).to.equal('/foo');
[C,test/listener.js,265]                 server.stop(done);
[C,test/listener.js,266]             };
[C,test/listener.js,267] 
[C,test/listener.js,268]             server.connection();
[C,test/listener.js,269]             server.register({ register: Nes, options: { auth: false } }, (err) => {
[C,test/listener.js,270] 
[C,test/listener.js,271]                 expect(err).to.not.exist();
[C,test/listener.js,272]                 server.connection();
[C,test/listener.js,273] 
[C,test/listener.js,274]                 server.subscription('/{params*}', { onSubscribe: onSubscribe, onUnsubscribe: onUnsubscribe });
[C,test/listener.js,275] 
[C,test/listener.js,276]                 server.start((err) => {
[C,test/listener.js,277] 
[A,test/listener.js,278]                     expect(err).to.not.exist();
[C,test/listener.js,279]                     client = new Nes.Client('http://localhost:' + server.connections[0].info.port);
[C,test/listener.js,280]                     client.connect(() => {
[C,test/listener.js,281] 
[C,test/listener.js,282]                         client.subscribe('/foo', Hoek.ignore, Hoek.ignore);
[C,test/listener.js,283]                     });
[C,test/listener.js,284]                 });
[C,test/listener.js,285]             });
[C,test/listener.js,286]         });
[C,test/listener.js,287]     });
[C,test/listener.js,288] 
[C,test/listener.js,289]     describe('publish()', () => {
[C,test/listener.js,290] 
[C,test/listener.js,291]         it('publishes to a parameterized path', (done) => {
[C,test/listener.js,292] 
[C,test/listener.js,293]             const server = new Hapi.Server();
[C,test/listener.js,294]             server.connection();
[C,test/listener.js,295] 
[C,test/listener.js,296]             server.register({ register: Nes, options: { auth: false } }, (err) => {
[C,test/listener.js,297] 
[C,test/listener.js,298]                 expect(err).to.not.exist();
[C,test/listener.js,299] 
[C,test/listener.js,300]                 server.subscription('/a/{id}');
[C,test/listener.js,301] 
[C,test/listener.js,302]                 server.start((err) => {
[C,test/listener.js,303] 
[A,test/listener.js,304]                     expect(err).to.not.exist();
[C,test/listener.js,305]                     const client = new Nes.Client('http://localhost:' + server.info.port);
[C,test/listener.js,306]                     client.connect((err) => {
[C,test/listener.js,307] 
[C,test/listener.js,308]                         expect(err).to.not.exist();
[C,test/listener.js,309] 
[C,test/listener.js,310]                         const handler = (update) => {
[C,test/listener.js,311] 
[C,test/listener.js,312]                             expect(update).to.equal('2');
[C,test/listener.js,313]                             client.disconnect();
[C,test/listener.js,314]                             server.stop(done);
[C,test/listener.js,315]                         };
[C,test/listener.js,316] 
[C,test/listener.js,317]                         client.subscribe('/a/b', handler, (err) => {
[C,test/listener.js,318] 
[C,test/listener.js,319]                             expect(err).to.not.exist();
@@ -322,30 +331,31 @@ describe('Listener', () => {
[C,test/listener.js,331]             server.connection();
[C,test/listener.js,332] 
[C,test/listener.js,333]             server.register({ register: Nes, options: { auth: false } }, (err) => {
[C,test/listener.js,334] 
[C,test/listener.js,335]                 expect(err).to.not.exist();
[C,test/listener.js,336] 
[C,test/listener.js,337]                 const filter = function (path, update, options, next) {
[C,test/listener.js,338] 
[C,test/listener.js,339]                     return next(update.a === 1);
[C,test/listener.js,340]                 };
[C,test/listener.js,341] 
[C,test/listener.js,342]                 server.subscription('/updates', { filter: filter });
[C,test/listener.js,343] 
[C,test/listener.js,344]                 server.start((err) => {
[C,test/listener.js,345] 
[A,test/listener.js,346]                     expect(err).to.not.exist();
[C,test/listener.js,347]                     const client = new Nes.Client('http://localhost:' + server.info.port);
[C,test/listener.js,348]                     client.connect((err) => {
[C,test/listener.js,349] 
[C,test/listener.js,350]                         expect(err).to.not.exist();
[C,test/listener.js,351] 
[C,test/listener.js,352]                         const handler = (update) => {
[C,test/listener.js,353] 
[C,test/listener.js,354]                             expect(update).to.deep.equal({ a: 1 });
[C,test/listener.js,355]                             client.disconnect();
[C,test/listener.js,356]                             server.stop(done);
[C,test/listener.js,357]                         };
[C,test/listener.js,358] 
[C,test/listener.js,359]                         client.subscribe('/updates', handler, (err) => {
[C,test/listener.js,360] 
[C,test/listener.js,361]                             expect(err).to.not.exist();
@@ -363,30 +373,31 @@ describe('Listener', () => {
[C,test/listener.js,373]             server.connection();
[C,test/listener.js,374] 
[C,test/listener.js,375]             server.register({ register: Nes, options: { auth: false } }, (err) => {
[C,test/listener.js,376] 
[C,test/listener.js,377]                 expect(err).to.not.exist();
[C,test/listener.js,378] 
[C,test/listener.js,379]                 const filter = function (path, update, options, next) {
[C,test/listener.js,380] 
[C,test/listener.js,381]                     return next(options.internal.b === 1);
[C,test/listener.js,382]                 };
[C,test/listener.js,383] 
[C,test/listener.js,384]                 server.subscription('/updates', { filter: filter });
[C,test/listener.js,385] 
[C,test/listener.js,386]                 server.start((err) => {
[C,test/listener.js,387] 
[A,test/listener.js,388]                     expect(err).to.not.exist();
[C,test/listener.js,389]                     const client = new Nes.Client('http://localhost:' + server.info.port);
[C,test/listener.js,390]                     client.connect((err) => {
[C,test/listener.js,391] 
[C,test/listener.js,392]                         expect(err).to.not.exist();
[C,test/listener.js,393] 
[C,test/listener.js,394]                         const handler = (update) => {
[C,test/listener.js,395] 
[C,test/listener.js,396]                             expect(update).to.deep.equal({ a: 1 });
[C,test/listener.js,397]                             client.disconnect();
[C,test/listener.js,398]                             server.stop(done);
[C,test/listener.js,399]                         };
[C,test/listener.js,400] 
[C,test/listener.js,401]                         client.subscribe('/updates', handler, (err) => {
[C,test/listener.js,402] 
[C,test/listener.js,403]                             expect(err).to.not.exist();
@@ -443,30 +454,31 @@ describe('Listener', () => {
[C,test/listener.js,454] 
[C,test/listener.js,455]     describe('_subscribe()', () => {
[C,test/listener.js,456] 
[C,test/listener.js,457]         it('subscribes to two paths on same subscription', (done) => {
[C,test/listener.js,458] 
[C,test/listener.js,459]             const server = new Hapi.Server();
[C,test/listener.js,460]             server.connection();
[C,test/listener.js,461]             server.register({ register: Nes, options: { auth: false } }, (err) => {
[C,test/listener.js,462] 
[C,test/listener.js,463]                 expect(err).to.not.exist();
[C,test/listener.js,464] 
[C,test/listener.js,465]                 server.subscription('/{id}', {});
[C,test/listener.js,466] 
[C,test/listener.js,467]                 server.start((err) => {
[C,test/listener.js,468] 
[A,test/listener.js,469]                     expect(err).to.not.exist();
[C,test/listener.js,470]                     const client = new Nes.Client('http://localhost:' + server.info.port);
[C,test/listener.js,471]                     client.connect(() => {
[C,test/listener.js,472] 
[C,test/listener.js,473]                         let called = false;
[C,test/listener.js,474]                         const handler1 = (update1) => {
[C,test/listener.js,475] 
[C,test/listener.js,476]                             called = true;
[C,test/listener.js,477]                         };
[C,test/listener.js,478] 
[C,test/listener.js,479]                         client.subscribe('/5', handler1, (err) => {
[C,test/listener.js,480] 
[C,test/listener.js,481]                             expect(err).to.not.exist();
[C,test/listener.js,482] 
[C,test/listener.js,483]                             const handler2 = (update2) => {
[C,test/listener.js,484] 
@@ -498,63 +510,67 @@ describe('Listener', () => {
[C,test/listener.js,510]             });
[C,test/listener.js,511]         });
[C,test/listener.js,512] 
[C,test/listener.js,513]         it('errors on double subscribe to same paths', (done) => {
[C,test/listener.js,514] 
[C,test/listener.js,515]             const server = new Hapi.Server();
[C,test/listener.js,516]             server.connection();
[C,test/listener.js,517]             server.register({ register: Nes, options: { auth: false } }, (err) => {
[C,test/listener.js,518] 
[C,test/listener.js,519]                 expect(err).to.not.exist();
[C,test/listener.js,520] 
[C,test/listener.js,521]                 server.subscription('/{id}', {});
[C,test/listener.js,522] 
[C,test/listener.js,523]                 server.start((err) => {
[C,test/listener.js,524] 
[A,test/listener.js,525]                     expect(err).to.not.exist();
[C,test/listener.js,526]                     const client = new Nes.Client('http://localhost:' + server.info.port);
[C,test/listener.js,527]                     client.connect(() => {
[C,test/listener.js,528] 
[C,test/listener.js,529]                         client.subscribe('/5', Hoek.ignore, (err) => {
[C,test/listener.js,530] 
[A,test/listener.js,531]                             expect(err).to.not.exist();
[C,test/listener.js,532]                             const request = {
[C,test/listener.js,533]                                 type: 'sub',
[C,test/listener.js,534]                                 path: '/5'
[C,test/listener.js,535]                             };
[C,test/listener.js,536] 
[C,test/listener.js,537]                             client._send(request, true, (err) => {
[C,test/listener.js,538] 
[A,test/listener.js,539]                                 expect(err).to.not.exist();
[C,test/listener.js,540]                                 client.disconnect();
[C,test/listener.js,541]                                 server.stop(done);
[C,test/listener.js,542]                             });
[C,test/listener.js,543]                         });
[C,test/listener.js,544]                     });
[C,test/listener.js,545]                 });
[C,test/listener.js,546]             });
[C,test/listener.js,547]         });
[C,test/listener.js,548] 
[C,test/listener.js,549]         it('errors on path with query', (done) => {
[C,test/listener.js,550] 
[C,test/listener.js,551]             const server = new Hapi.Server();
[C,test/listener.js,552]             server.connection();
[C,test/listener.js,553]             server.register({ register: Nes, options: { auth: false } }, (err) => {
[C,test/listener.js,554] 
[C,test/listener.js,555]                 expect(err).to.not.exist();
[C,test/listener.js,556] 
[C,test/listener.js,557]                 server.subscription('/');
[C,test/listener.js,558] 
[C,test/listener.js,559]                 server.start((err) => {
[C,test/listener.js,560] 
[A,test/listener.js,561]                     expect(err).to.not.exist();
[C,test/listener.js,562]                     const client = new Nes.Client('http://localhost:' + server.info.port);
[C,test/listener.js,563]                     client.connect((err) => {
[C,test/listener.js,564] 
[C,test/listener.js,565]                         expect(err).to.not.exist();
[C,test/listener.js,566]                         client.subscribe('/?5', Hoek.ignore, (err) => {
[C,test/listener.js,567] 
[C,test/listener.js,568]                             expect(err).to.exist();
[C,test/listener.js,569]                             expect(err.message).to.equal('Subscription path cannot contain query');
[C,test/listener.js,570] 
[C,test/listener.js,571]                             client.disconnect();
[C,test/listener.js,572]                             server.stop(done);
[C,test/listener.js,573]                         });
[C,test/listener.js,574]                     });
[C,test/listener.js,575]                 });
[C,test/listener.js,576]             });
@@ -573,96 +589,100 @@ describe('Listener', () => {
[C,test/listener.js,589]             }, options);
[C,test/listener.js,590]             return seen;
[C,test/listener.js,591]         };
[C,test/listener.js,592] 
[C,test/listener.js,593]         it('returns connected sockets', (done) => {
[C,test/listener.js,594] 
[C,test/listener.js,595]             const server = new Hapi.Server();
[C,test/listener.js,596]             server.connection();
[C,test/listener.js,597] 
[C,test/listener.js,598]             server.register({ register: Nes, options: { auth: false } }, (err) => {
[C,test/listener.js,599] 
[C,test/listener.js,600]                 expect(err).to.not.exist();
[C,test/listener.js,601] 
[C,test/listener.js,602]                 server.start((err) => {
[C,test/listener.js,603] 
[A,test/listener.js,604]                     expect(err).to.not.exist();
[C,test/listener.js,605]                     const client = new Nes.Client('http://localhost:' + server.info.port);
[C,test/listener.js,606]                     client.connect((err) => {
[C,test/listener.js,607] 
[C,test/listener.js,608]                         expect(err).to.not.exist();
[C,test/listener.js,609]                         expect(countSockets(server)).to.equal(1);
[C,test/listener.js,610] 
[C,test/listener.js,611]                         server.stop(done);
[C,test/listener.js,612]                     });
[C,test/listener.js,613]                 });
[C,test/listener.js,614]             });
[C,test/listener.js,615]         });
[C,test/listener.js,616] 
[C,test/listener.js,617]         it('returns sockets on a subscription', (done) => {
[C,test/listener.js,618] 
[C,test/listener.js,619]             const server = new Hapi.Server();
[C,test/listener.js,620]             server.connection();
[C,test/listener.js,621] 
[C,test/listener.js,622]             server.register({ register: Nes, options: { auth: false } }, (err) => {
[C,test/listener.js,623] 
[C,test/listener.js,624]                 expect(err).to.not.exist();
[C,test/listener.js,625] 
[C,test/listener.js,626]                 server.subscription('/a/{id}');
[C,test/listener.js,627]                 server.subscription('/b');
[C,test/listener.js,628] 
[C,test/listener.js,629]                 server.start((err) => {
[C,test/listener.js,630] 
[A,test/listener.js,631]                     expect(err).to.not.exist();
[C,test/listener.js,632]                     let client = new Nes.Client('http://localhost:' + server.info.port);
[C,test/listener.js,633]                     client.connect((err) => {
[C,test/listener.js,634] 
[C,test/listener.js,635]                         expect(err).to.not.exist();
[C,test/listener.js,636] 
[C,test/listener.js,637]                         client.subscribe('/b', Hoek.ignore, Hoek.ignore);
[C,test/listener.js,638] 
[C,test/listener.js,639]                         client = new Nes.Client('http://localhost:' + server.info.port);
[C,test/listener.js,640]                         client.connect((err) => {
[C,test/listener.js,641] 
[C,test/listener.js,642]                             expect(err).to.not.exist();
[C,test/listener.js,643]                             client.subscribe('/a/b', Hoek.ignore, (err) => {
[C,test/listener.js,644] 
[A,test/listener.js,645]                                 expect(err).to.not.exist();
[C,test/listener.js,646]                                 expect(countSockets(server)).to.equal(2);
[C,test/listener.js,647]                                 expect(countSockets(server, { subscription: '/a/a' })).to.equal(0);
[C,test/listener.js,648]                                 expect(countSockets(server, { subscription: '/a/b' })).to.equal(1);
[C,test/listener.js,649] 
[C,test/listener.js,650]                                 expect(countSockets(server, { subscription: '/b' })).to.equal(1);
[C,test/listener.js,651] 
[C,test/listener.js,652]                                 expect(countSockets(server, { subscription: '/foo' })).to.equal(0);
[C,test/listener.js,653] 
[C,test/listener.js,654]                                 server.stop(done);
[C,test/listener.js,655]                             });
[C,test/listener.js,656]                         });
[C,test/listener.js,657]                     });
[C,test/listener.js,658]                 });
[C,test/listener.js,659]             });
[C,test/listener.js,660]         });
[C,test/listener.js,661] 
[C,test/listener.js,662]         it('ignores not participating connections', (done) => {
[C,test/listener.js,663] 
[C,test/listener.js,664]             const server = new Hapi.Server();
[C,test/listener.js,665]             server.connection();
[C,test/listener.js,666] 
[C,test/listener.js,667]             server.register({ register: Nes, options: { auth: false } }, (err) => {
[C,test/listener.js,668] 
[C,test/listener.js,669]                 expect(err).to.not.exist();
[C,test/listener.js,670] 
[C,test/listener.js,671]                 server.start((err) => {
[C,test/listener.js,672] 
[A,test/listener.js,673]                     expect(err).to.not.exist();
[C,test/listener.js,674]                     const client = new Nes.Client('http://localhost:' + server.info.port);
[C,test/listener.js,675]                     client.connect((err) => {
[C,test/listener.js,676] 
[C,test/listener.js,677]                         expect(err).to.not.exist();
[C,test/listener.js,678] 
[C,test/listener.js,679]                         server.connection();
[C,test/listener.js,680]                         expect(countSockets(server)).to.equal(1);
[C,test/listener.js,681] 
[C,test/listener.js,682]                         server.stop(done);
[C,test/listener.js,683]                     });
[C,test/listener.js,684]                 });
[C,test/listener.js,685]             });
[C,test/listener.js,686]         });
[C,test/listener.js,687]     });
[C,test/listener.js,688] 
@@ -40,30 +40,31 @@ describe('Socket', () => {
[C,test/socket.js,40]         server.register({ register: Nes, options: { onConnection: onConnection, auth: false } }, (err) => {
[C,test/socket.js,41] 
[C,test/socket.js,42]             expect(err).to.not.exist();
[C,test/socket.js,43] 
[C,test/socket.js,44]             server.route({
[C,test/socket.js,45]                 method: 'GET',
[C,test/socket.js,46]                 path: '/',
[C,test/socket.js,47]                 handler: function (request, reply) {
[C,test/socket.js,48] 
[C,test/socket.js,49]                     return reply(request.socket.app.x);
[C,test/socket.js,50]                 }
[C,test/socket.js,51]             });
[C,test/socket.js,52] 
[C,test/socket.js,53]             server.start((err) => {
[C,test/socket.js,54] 
[A,test/socket.js,55]                 expect(err).to.not.exist();
[C,test/socket.js,56]                 client = new Nes.Client('http://localhost:' + server.info.port);
[C,test/socket.js,57]                 client.connect(() => {
[C,test/socket.js,58] 
[C,test/socket.js,59]                     client.request('/', (err, payload, statusCode, headers) => {
[C,test/socket.js,60] 
[C,test/socket.js,61]                         expect(err).to.not.exist();
[C,test/socket.js,62]                         expect(payload).to.equal('hello');
[C,test/socket.js,63]                         expect(statusCode).to.equal(200);
[C,test/socket.js,64] 
[C,test/socket.js,65]                         client.disconnect();
[C,test/socket.js,66]                         server.stop(done);
[C,test/socket.js,67]                     });
[C,test/socket.js,68]                 });
[C,test/socket.js,69]             });
[C,test/socket.js,70]         });
@@ -74,63 +75,65 @@ describe('Socket', () => {
[C,test/socket.js,75]         it('closes connection', (done) => {
[C,test/socket.js,76] 
[C,test/socket.js,77]             const onMessage = function (socket, message, reply) {
[C,test/socket.js,78] 
[C,test/socket.js,79]                 socket.disconnect();
[C,test/socket.js,80]             };
[C,test/socket.js,81] 
[C,test/socket.js,82]             const server = new Hapi.Server();
[C,test/socket.js,83]             server.connection();
[C,test/socket.js,84]             server.register({ register: Nes, options: { onMessage: onMessage } }, (err) => {
[C,test/socket.js,85] 
[C,test/socket.js,86]                 expect(err).to.not.exist();
[C,test/socket.js,87] 
[C,test/socket.js,88]                 server.start((err) => {
[C,test/socket.js,89] 
[A,test/socket.js,90]                     expect(err).to.not.exist();
[C,test/socket.js,91]                     const client = new Nes.Client('http://localhost:' + server.info.port);
[C,test/socket.js,92]                     client.onDisconnect = function () {
[C,test/socket.js,93] 
[C,test/socket.js,94]                         client.disconnect();
[C,test/socket.js,95]                         server.stop(done);
[C,test/socket.js,96]                     };
[C,test/socket.js,97] 
[C,test/socket.js,98]                     client.connect(() => {
[C,test/socket.js,99] 
[D,test/socket.js,98]                         client.message('winning', (err, response) => { });
[A,test/socket.js,100]                         client.message('winning', (errIgnore, response) => { });
[C,test/socket.js,101]                     });
[C,test/socket.js,102]                 });
[C,test/socket.js,103]             });
[C,test/socket.js,104]         });
[C,test/socket.js,105]     });
[C,test/socket.js,106] 
[C,test/socket.js,107]     describe('send()', () => {
[C,test/socket.js,108] 
[C,test/socket.js,109]         it('sends custom message', (done) => {
[C,test/socket.js,110] 
[C,test/socket.js,111]             const onConnection = function (socket) {
[C,test/socket.js,112] 
[C,test/socket.js,113]                 socket.send('goodbye');
[C,test/socket.js,114]             };
[C,test/socket.js,115] 
[C,test/socket.js,116]             const server = new Hapi.Server();
[C,test/socket.js,117]             server.connection();
[C,test/socket.js,118]             server.register({ register: Nes, options: { onConnection: onConnection } }, (err) => {
[C,test/socket.js,119] 
[C,test/socket.js,120]                 expect(err).to.not.exist();
[C,test/socket.js,121] 
[C,test/socket.js,122]                 server.start((err) => {
[C,test/socket.js,123] 
[A,test/socket.js,124]                     expect(err).to.not.exist();
[C,test/socket.js,125]                     const client = new Nes.Client('http://localhost:' + server.info.port);
[C,test/socket.js,126]                     client.onUpdate = function (message) {
[C,test/socket.js,127] 
[C,test/socket.js,128]                         expect(message).to.equal('goodbye');
[C,test/socket.js,129]                         client.disconnect();
[C,test/socket.js,130]                         server.stop(done);
[C,test/socket.js,131]                     };
[C,test/socket.js,132] 
[C,test/socket.js,133]                     client.connect(() => { });
[C,test/socket.js,134]                 });
[C,test/socket.js,135]             });
[C,test/socket.js,136]         });
[C,test/socket.js,137]     });
[C,test/socket.js,138] 
[C,test/socket.js,139]     describe('_send()', () => {
@@ -141,30 +144,31 @@ describe('Socket', () => {
[C,test/socket.js,144]             let client;
[C,test/socket.js,145]             server.connection();
[C,test/socket.js,146]             server.register({ register: Nes, options: { auth: false } }, (err) => {
[C,test/socket.js,147] 
[C,test/socket.js,148]                 expect(err).to.not.exist();
[C,test/socket.js,149] 
[C,test/socket.js,150]                 server.on('log', (event, tags) => {
[C,test/socket.js,151] 
[C,test/socket.js,152]                     expect(event.data).to.equal('other');
[C,test/socket.js,153]                     client.disconnect();
[C,test/socket.js,154]                     server.stop(done);
[C,test/socket.js,155]                 });
[C,test/socket.js,156] 
[C,test/socket.js,157]                 server.start((err) => {
[C,test/socket.js,158] 
[A,test/socket.js,159]                     expect(err).to.not.exist();
[C,test/socket.js,160]                     client = new Nes.Client('http://localhost:' + server.info.port);
[C,test/socket.js,161]                     client.connect((err) => {
[C,test/socket.js,162] 
[C,test/socket.js,163]                         expect(err).to.not.exist();
[C,test/socket.js,164]                         const a = { id: 1, type: 'other' };
[C,test/socket.js,165]                         a.c = a;                    // Circular reference
[C,test/socket.js,166] 
[C,test/socket.js,167]                         server.connections[0].plugins.nes._listener._sockets.forEach((socket) => {
[C,test/socket.js,168] 
[C,test/socket.js,169]                             socket._send(a);
[C,test/socket.js,170]                         });
[C,test/socket.js,171]                     });
[C,test/socket.js,172]                 });
[C,test/socket.js,173]             });
[C,test/socket.js,174]         });
@@ -182,30 +186,31 @@ describe('Socket', () => {
[C,test/socket.js,186] 
[C,test/socket.js,187]                 server.route({
[C,test/socket.js,188]                     method: 'GET',
[C,test/socket.js,189]                     path: '/',
[C,test/socket.js,190]                     config: {
[C,test/socket.js,191]                         id: 'resource',
[C,test/socket.js,192]                         handler: function (request, reply) {
[C,test/socket.js,193] 
[C,test/socket.js,194]                             return reply('hello');
[C,test/socket.js,195]                         }
[C,test/socket.js,196]                     }
[C,test/socket.js,197]                 });
[C,test/socket.js,198] 
[C,test/socket.js,199]                 server.start((err) => {
[C,test/socket.js,200] 
[A,test/socket.js,201]                     expect(err).to.not.exist();
[C,test/socket.js,202]                     const client = new Nes.Client('http://localhost:' + server.info.port);
[C,test/socket.js,203]                     client.connect(() => {
[C,test/socket.js,204] 
[C,test/socket.js,205]                         client.request('resource', (err, payload, statusCode, headers) => {
[C,test/socket.js,206] 
[C,test/socket.js,207]                             expect(err).to.not.exist();
[C,test/socket.js,208]                             expect(payload).to.equal('hello');
[C,test/socket.js,209]                             expect(statusCode).to.equal(200);
[C,test/socket.js,210] 
[C,test/socket.js,211]                             client.disconnect();
[C,test/socket.js,212]                             server.stop(done);
[C,test/socket.js,213]                         });
[C,test/socket.js,214]                     });
[C,test/socket.js,215]                 });
[C,test/socket.js,216]             });
@@ -221,30 +226,31 @@ describe('Socket', () => {
[C,test/socket.js,226] 
[C,test/socket.js,227]                 server.route({
[C,test/socket.js,228]                     method: 'GET',
[C,test/socket.js,229]                     path: '/',
[C,test/socket.js,230]                     config: {
[C,test/socket.js,231]                         id: 'resource',
[C,test/socket.js,232]                         handler: function (request, reply) {
[C,test/socket.js,233] 
[C,test/socket.js,234]                             return reply('hello');
[C,test/socket.js,235]                         }
[C,test/socket.js,236]                     }
[C,test/socket.js,237]                 });
[C,test/socket.js,238] 
[C,test/socket.js,239]                 server.start((err) => {
[C,test/socket.js,240] 
[A,test/socket.js,241]                     expect(err).to.not.exist();
[C,test/socket.js,242]                     const client = new Nes.Client('http://localhost:' + server.info.port);
[C,test/socket.js,243]                     client.connect(() => {
[C,test/socket.js,244] 
[C,test/socket.js,245]                         client.request('something', (err, payload, statusCode, headers) => {
[C,test/socket.js,246] 
[C,test/socket.js,247]                             expect(err).to.exist();
[C,test/socket.js,248]                             expect(statusCode).to.equal(404);
[C,test/socket.js,249] 
[C,test/socket.js,250]                             client.disconnect();
[C,test/socket.js,251]                             server.stop(done);
[C,test/socket.js,252]                         });
[C,test/socket.js,253]                     });
[C,test/socket.js,254]                 });
[C,test/socket.js,255]             });
[C,test/socket.js,256]         });
@@ -259,30 +265,31 @@ describe('Socket', () => {
[C,test/socket.js,265] 
[C,test/socket.js,266]                 server.route({
[C,test/socket.js,267]                     method: '*',
[C,test/socket.js,268]                     path: '/',
[C,test/socket.js,269]                     config: {
[C,test/socket.js,270]                         id: 'resource',
[C,test/socket.js,271]                         handler: function (request, reply) {
[C,test/socket.js,272] 
[C,test/socket.js,273]                             return reply('hello');
[C,test/socket.js,274]                         }
[C,test/socket.js,275]                     }
[C,test/socket.js,276]                 });
[C,test/socket.js,277] 
[C,test/socket.js,278]                 server.start((err) => {
[C,test/socket.js,279] 
[A,test/socket.js,280]                     expect(err).to.not.exist();
[C,test/socket.js,281]                     const client = new Nes.Client('http://localhost:' + server.info.port);
[C,test/socket.js,282]                     client.connect(() => {
[C,test/socket.js,283] 
[C,test/socket.js,284]                         client.request('resource', (err, payload, statusCode, headers) => {
[C,test/socket.js,285] 
[C,test/socket.js,286]                             expect(err).to.exist();
[C,test/socket.js,287]                             expect(statusCode).to.equal(400);
[C,test/socket.js,288] 
[C,test/socket.js,289]                             client.disconnect();
[C,test/socket.js,290]                             server.stop(done);
[C,test/socket.js,291]                         });
[C,test/socket.js,292]                     });
[C,test/socket.js,293]                 });
[C,test/socket.js,294]             });
[C,test/socket.js,295]         });
@@ -294,30 +301,31 @@ describe('Socket', () => {
[C,test/socket.js,301]             server.register({ register: Nes, options: { auth: false } }, (err) => {
[C,test/socket.js,302] 
[C,test/socket.js,303]                 expect(err).to.not.exist();
[C,test/socket.js,304] 
[C,test/socket.js,305]                 server.route({
[C,test/socket.js,306]                     method: 'GET',
[C,test/socket.js,307]                     path: '/',
[C,test/socket.js,308]                     handler: function (request, reply) {
[C,test/socket.js,309] 
[C,test/socket.js,310]                         return reply('hello');
[C,test/socket.js,311]                     }
[C,test/socket.js,312]                 });
[C,test/socket.js,313] 
[C,test/socket.js,314]                 server.start((err) => {
[C,test/socket.js,315] 
[A,test/socket.js,316]                     expect(err).to.not.exist();
[C,test/socket.js,317]                     const client = new Ws('http://localhost:' + server.info.port);
[C,test/socket.js,318] 
[C,test/socket.js,319]                     client.on('message', (data, flags) => {
[C,test/socket.js,320] 
[C,test/socket.js,321]                         const message = JSON.parse(data);
[C,test/socket.js,322]                         expect(message.payload).to.deep.equal({
[C,test/socket.js,323]                             error: 'Bad Request',
[C,test/socket.js,324]                             message: 'Cannot parse message'
[C,test/socket.js,325]                         });
[C,test/socket.js,326] 
[C,test/socket.js,327]                         expect(message.statusCode).to.equal(400);
[C,test/socket.js,328] 
[C,test/socket.js,329]                         client.close();
[C,test/socket.js,330]                         server.stop(done);
[C,test/socket.js,331]                     });
@@ -331,64 +339,67 @@ describe('Socket', () => {
[C,test/socket.js,339]                     });
[C,test/socket.js,340]                 });
[C,test/socket.js,341]             });
[C,test/socket.js,342]         });
[C,test/socket.js,343] 
[C,test/socket.js,344]         it('errors on auth endpoint request', (done) => {
[C,test/socket.js,345] 
[C,test/socket.js,346]             const server = new Hapi.Server();
[C,test/socket.js,347]             server.connection();
[C,test/socket.js,348]             server.register({ register: Nes, options: { auth: { password: 'password' } } }, (err) => {
[C,test/socket.js,349] 
[C,test/socket.js,350]                 expect(err).to.not.exist();
[C,test/socket.js,351] 
[C,test/socket.js,352]                 server.start((err) => {
[C,test/socket.js,353] 
[A,test/socket.js,354]                     expect(err).to.not.exist();
[C,test/socket.js,355]                     const client = new Nes.Client('http://localhost:' + server.info.port);
[C,test/socket.js,356]                     client.connect((err) => {
[C,test/socket.js,357] 
[C,test/socket.js,358]                         expect(err).to.not.exist();
[C,test/socket.js,359]                         client.request('/nes/auth', (err, payload, statusCode, headers) => {
[C,test/socket.js,360] 
[A,test/socket.js,361]                             expect(err).to.exist();
